### ISSUE TITLE
error: no such subcommand: `+bpf`

### ISSUE THREAD

**Issue by geofflangenderfer** [solana-labs/example-helloworld#444]
not working on aarch64 mac
```
solana_hello_world 
❯ npm run build:program-rust


> helloworld@0.0.1 build:program-rust
> cargo build-bpf --manifest-path=./src/program-rust/Cargo.toml --bpf-out-dir=dist/program

BPF SDK: /Users/lgeoff/.local/share/solana/install/releases/1.10.5/solana-release/bin/sdk/
bpf
cargo-build-bpf child: rustup toolchain list -v
cargo-build-bpf child: rustup toolchain link bpf /Users/lgeoff/.local/share/solana/install
/releases/1.10.5/solana-release/bin/sdk/bpf/dependencies/bpf-tools/rust
cargo-build-bpf child: cargo +bpf build --target bpfel-unknown-unknown --release
error: no such subcommand: `+bpf`

        Did you mean `b`?


```
```
solana_hello_world on  master is 📦 v0.0.1 via  v17.5.0 on ☁️  (us-west-2)
❯ rustc --version
rustc 1.59.0

solana_hello_world on  master is 📦 v0.0.1 via  v17.5.0 on ☁️  (us-west-2)
❯ rustup --version
rustup 1.24.3 (ce5817a94 2021-05-31)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.59.0 (9d1b2106e 2022-02-23)`

solana_hello_world on  master is 📦 v0.0.1 via  v17.5.0 on ☁️  (us-west-2)
❯
```

**Comment by septemhill** [comment]
I have the samp problem, even use rust nightly version. It seems need to install `https://crates.io/crates/solana-cargo-build-bpf`, but cargo install `solana-cargo-build-bpf` cannot find this.

**Comment by rollschild** [comment]
I'm seeing a similar issue here. Any help would be appreciated!

**Comment by geofflangenderfer** [comment]
[This crate has been yanked, but it is still available for download for other crates that may be depending on it.](https://crates.io/crates/solana-cargo-build-bpf)

**Comment by geofflangenderfer** [comment]
working!
```
example-helloworld 
❯ solana --version
solana-cli 1.10.7 (src:8bce2dd4; feat:1122441720)

example-helloworld 
❯ rustc --version
rustc 1.60.0 (7737e0b5c 2022-04-04)

example-helloworld 
❯ npm run build:program-rust
     0K                                    100%  760K=0.2s

2022-04-09 14:01:03 (760 KB/s) - ‘criterion-v2.3.2-osx-x86_64.tar.bz2’ saved [180979/180979]

+ tar --strip-components 1 -jxf criterion-v2.3.2-osx-x86_64.tar.bz2
+ ./bpf-tools/rust/bin/rustc --version
+ ./bpf-tools/rust/bin/rustc --print sysroot
+ set +e
+ rustup toolchain uninstall bpf
info: uninstalling toolchain 'bpf'
info: toolchain 'bpf' uninstalled
+ set -e
+ rustup toolchain link bpf bpf-tools/rust
+ exit 0
cargo-build-bpf child: /Users/lgeoff/.local/share/solana/install/releases/1.10.7/solana-release/bin/sdk/bpf/dependencies/bpf-tools/llvm/bin/llvm-readelf --dyn-symbols /Users/lgeoff/work/example-helloworld/dist/program/helloworld.so

To deploy this program:
  $ solana program deploy /Users/lgeoff/work/example-helloworld/dist/program/helloworld.so
The program address will default to this keypair (override with --program-id):
  /Users/lgeoff/work/example-helloworld/dist/program/helloworld-keypair.json

example-helloworld 
❯ 
```

**Comment by rollschild** [comment]
@geofflangenderfer Hey, could you please explain _how_ did you make it work? Because in your previous comment you mentioned "Installing older version doesn't work" but later you said "working". I'm really curious how you made it work? Where/how did you install the `build-bpf`? Thank you so much!

**Comment by geofflangenderfer** [comment]
everything worked as expected with those versions of rustc and solana.

How would I reproduce this from scratch?:
- install rustc 1.6, which is current stable version
- install solana 1.10.7. You can try [pre-compiled release](https://github.com/solana-labs/solana/releases) or build from source if necessary
- at this point,`npm run build:program-rust` just worked

I was slightly curious to see if I could find any change in recent commits that would've fixed the bpf build issue. I couldn't find anything.

**Comment by jstarry** [comment]
> Where/how did you install the build-bpf?

This is installed as `cargo-build-bpf` when [installing the precompiled solana tool suite](https://docs.solana.com/cli/install-solana-cli-tools) or using `scripts/cargo-install-all.sh` when building from source. And just a note that cargo subcommands like `cargo build-bpf` are prefixed by `cargo-`, I think this is probably why `solana-cargo-build-bpf` was yanked, but not sure. Anyways you can test if the subcommand is setup properly by ensuring `~/.local/share/solana/install/active_release/bin` is in your PATH and running the following:
```bash
which cargo-build-bpf
~/.local/share/solana/install/active_release/bin/cargo-build-bpf
```

**Comment by geofflangenderfer** [comment]
install script did not change..
```
solana on  master via 🦀 v1.60.0 on ☁️  (us-west-2) took 14s
❯ git diff remotes/origin/v1.9 scripts/cargo-install-all.sh

solana on  master via 🦀 v1.60.0 on ☁️  (us-west-2)
❯
```

**Comment by jstarry** [comment]
> install script did not change..

Oh I wasn't commenting on the change in my reply. But to address that, M1 support didn't get added to source / precompiled binaries until a few months ago so most issues are probably a result of using older binariess

**Comment by geofflangenderfer** [comment]
just thinking out loud, thanks for your help @jstarry !

**Comment by x-N0** [comment]
Following @jstarry's answer, I'll like to add that:
You have to make sure that the path is added in the right shell configuration file, if not, adding Solana to the PATH will work until you exit the shell.

[Locate which config file](https://stackoverflow.com/questions/11629304/locate-my-bash-configuration-file) you're using.

Hope it helps.

**Comment by Mort1B** [comment]
I solve this problem by installing stable version of solana CLI Suite. 

`sh -c "$(curl -sSfL https://release.solana.com/stable/install)"`

**Comment by acamill** [comment]
I get this issue each time I reinstall, and each time it's because I forget to do 
```
Adding 
export PATH="/Users/acammm/.local/share/solana/install/active_release/bin:$PATH" to /Users/acammm/.profile
Adding 
export PATH="/Users/acammm/.local/share/solana/install/active_release/bin:$PATH" to /Users/acammm/.zprofile
```
After installing Solana

**Comment by adrianperreault** [comment]
This fixed it for me (on an M1 macOS 13.0) : 
```
brew uninstall rust
 curl https://sh.rustup.rs -sSf | sh
```
